pandas/_libs/intervaltree.pxi.in

```diff
@@ -81,7 +81,9 @@ cdef class IntervalTree(IntervalMixin):
         """How to sort the left labels; this is used for binary search
         """
         if self._left_sorter is None:
-            self._left_sorter = np.argsort(self.left)
+            left_right = np.asarray([(self.left[i], self.right[i]) for i in range(0,
```

Comment:
this looks likely to be non-performant.  can we maybe check for uniqueness and only do this in the non-unique case?

Reply:
Sounds like a more decent solution. Actually, I'm a first-time contributor here, so I'm wondering what to do next. Do I have to close the PR?

Reply:
You can push a new commit to update your pull request with the changes

Reply:
i think elsewhere in the file we use `np.lexsort([self.left, self.right])` which should be similar to this

Reply:
Sound like a much better way for sorting, and I have made these changes in my latest commit.